test/e2e/util.go

```diff
@@ -1142,7 +1142,11 @@ func SSH(cmd, host, provider string) (string, string, int, error) {
 		return "", "", 0, fmt.Errorf("error getting signer for provider %s: '%v'", provider, err)
 	}
 
-	return util.RunSSHCommand(cmd, host, signer)
+	user := os.Getenv("KUBE_SSH_USER")
+	if user == "" {
```

Comment:
Yeah, I think RunSSHCommand should work like ssh - make an assumption if
not specified.

On Thu, Jun 18, 2015 at 6:29 PM, Justin Santa Barbara <
notifications@github.com> wrote:

> In test/e2e/util.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/9905#discussion_r32795268>
> :
>
> > @@ -1142,7 +1142,11 @@ func SSH(cmd, host, provider string) (string, string, int, error) {
> >  		return "", "", 0, fmt.Errorf("error getting signer for provider %s: '%v'", provider, err)
> >  	}
> >
> > -	return util.RunSSHCommand(cmd, host, signer)
> > +	user := os.Getenv("KUBE_SSH_USER")
> > +	if user == "" {
>
> util is supposed to be pretty generic I thought, so my feeling was that it
> shouldn't really be looking at the environment, and it would be even more
> questionable to have this if statement in there. I'm happy to just move
> this into RunSSHCommand (it would simplify the code) if you disagree.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/9905/files#r32795268>
> .
>